ddl/index.go

```diff
@@ -1284,6 +1302,95 @@ func (w *addIndexWorker) batchCheckUniqueKey(txn kv.Transaction, idxRecords []*i
 	return nil
 }
 
+func (w *addIndexWorker) backfillDataInTxnByRead(handleRange reorgBackfillTask) (taskCtx backfillTaskContext, errInTxn error) {
+	oprStartTime := time.Now()
+
+	txn, err := w.sessCtx.GetStore().Begin()
+	if err != nil {
+		return taskCtx, errors.Trace(err)
+	}
+
+	taskCtx.addedCount = 0
+	taskCtx.scanCount = 0
+	txn.SetOption(kv.Priority, w.priority)
+
+	idxRecords, nextKey, taskDone, err := w.fetchRowColVals(txn, handleRange)
+	if err != nil {
+		return taskCtx, errors.Trace(err)
+	}
+	taskCtx.nextKey = nextKey
+	taskCtx.done = taskDone
+	// TODO: delete later;
+	var minKey = make([]byte, len(nextKey)+2)
+	var maxKey = make([]byte, len(nextKey)+2)
+	// var prevKey = make([]byte, len(nextKey)+2)
+	// isSame := false
+	copy(minKey, nextKey)
+	copy(maxKey, nextKey)
+
+	err = w.batchCheckUniqueKey(txn, idxRecords)
```

Comment:
Add a TODO and delete this part here. This can't work for lightning because we haven't committed data here.